eth_utils/string.py

```diff
@@ -1,15 +1,33 @@
-import functools
 import codecs
+import functools
+import warnings
 
 from .types import (
     is_bytes,
-    is_text,
-    is_string,
     is_dict,
     is_list_like,
+    is_string,
+    is_text,
 )
 
 
+def _deprecated(fn):
+    @functools.wraps(fn)
+    def inner(*args, **kwargs):
+        warnings.simplefilter('always', DeprecationWarning)
+        warnings.warn(DeprecationWarning(
+            "The `{0}` function has been deprecated and will be removed in a "
+            "subsequent release of the eth-utils library. UTF8 cannot encode "
+            "some byte values in the 0-255 range which makes naive coersion between "
+            "bytes and text representations impossible without explicitly "
+            "declared encodings.".format(fn.__name__)
+        ))
+        warnings.resetwarnings()
+        return fn(*args, **kwargs)
+    return inner
+
+
+@_deprecated
```

Comment:
This deprecation warning should be submitted as a separate PR against the v0 branch. They can just be deleted in v1 (or if all usages aren't removed in this PR, then left in only until the cleanup is done).

Reply:
Made the pr to v0 branch. I plan to submit the cleanup prs 1 by 1 - and will delete them on the last one.